openshift/console
# ODC-7288: Add proxy to invoke knative services with a public route

**Fixes**: 
https://issues.redhat.com/browse/ODC-7288

Continued Work from #12693 
( Thanks @jerolimov for formulating the basic structure of the code )

**Solution Description**: 
Backend for Test Serverless Functions. 

Additions in this PR:
- Support for CloudEvents
- Updated API Design to establish a connection with Frontend
- Code Clean-up

**Screen shots / Gifs for design review**: 


**POSTMAN API TESTING**

[Screencast from 2023-05-11 03-07-45.webm](https://github.com/openshift/console/assets/47265560/756a112f-d24c-4c60-8300-82a6f3155af6)


**Unit test coverage report**: 

Added Unit Tests

![image](https://github.com/openshift/console/assets/47265560/b87063df-1647-4452-a914-41a8463b203b)


**Test setup:**

1. Install Serverless Operator and Serving Instance
2. Create the Functions from Console:
    * HTTP: https://github.com/openshift-dev-console/kn-func-typescript-http
    * CE: https://github.com/openshift-dev-console/kn-func-typescript-cloudevents
3. Import the Postman Collection for testing the APIs.
https://drive.google.com/file/d/1nlCimr_OxrmClSBSEs1fifM4ObQ2b9dc/view?usp=share_link
4. Make sure to change the POST URLs to add ur KSVC name/ns.
5. Send the POST Requests to get the result.



**Browser conformance**:
